pkg/sql/conn_executor.go

```diff
@@ -3687,3 +3759,56 @@ func init() {
 		return planGistFromCtx(ctx)
 	})
 }
+
+type jobsInfo struct {
+	jobsCreated           jobsCollection
+	uniqueJobsToCreate    map[descpb.ID]*jobs.Record
+	nonUniqueJobsToCreate []*jobs.Record
+}
+
+func newJobsInfo() *jobsInfo {
+	ret := &jobsInfo{}
+	ret.reset()
+	return ret
+}
+
+func (j *jobsInfo) addCreatedJobID(jobID ...jobspb.JobID) {
+	j.jobsCreated.add(jobID...)
+}
+
+func (j *jobsInfo) addNonUniqueJobToCreate(jobRecord *jobs.Record) {
+	j.nonUniqueJobsToCreate = append(j.nonUniqueJobsToCreate, jobRecord)
+}
+
+func (j *jobsInfo) reset() {
+	j.jobsCreated.reset()
+	j.uniqueJobsToCreate = make(map[descpb.ID]*jobs.Record)
+	j.nonUniqueJobsToCreate = nil
+}
+
+func (j *jobsInfo) numJobsToCreate() int {
+	return len(j.uniqueJobsToCreate) + len(j.nonUniqueJobsToCreate)
+}
+
+func (j *jobsInfo) hasJobsToCreate() bool {
+	return j.numJobsToCreate() > 0
+}
+
+func (j *jobsInfo) forEachJobToCreate(fn func(jobRecord *jobs.Record) error) error {
```

Comment:
nit: i think you can remove the word `jobs` from all these methods and improve readability. `x.jobs.forEachToCreate`, `x.jobs.numToCreate`, `x.jobs.hasAnyToCreate`? That's just my taste.

Reply:
done.